src/java/org/pantsbuild/tools/junit/impl/ScalaTestUtil.java

```diff
@@ -18,14 +23,40 @@ private ScalaTestUtil() {}
     }
   }
 
+  private static class ScalaTestJunitRunnerWrapper extends Runner implements Filterable {
```

Comment:
Did you consider upstreaming the Filterable implementation directly to scalatestplus-junit?: https://github.com/scalatest/scalatestplus-junit/blob/master/src/main/scala/org/scalatestplus/junit/JUnitRunner.scala

I'm fine with this as a temporary workaround that we TODO deletion of, but it feels a bit strange bolting this onto scalatest. I could see, at a higher layer in our runner, ensuring each `Runner` is `Filterable` and, if not, wrapping it in something like this when sharding o rother test filtering is turned on, but hacking it directly on to scalatest seems a bit arbitrary - we still permit sharding to silently fail for any other non Filterable Runners that may crop up.

Reply:
I added a comment to explain this, and posted a PR in scalatestplus-junit: 
https://github.com/scalatest/scalatestplus-junit/pull/8

If it's ok for you otherwise, It would help us very much to merge this temporarily fix, as we would like to use test sharding for scala tests

Reply:
I would like to add that there is no release version of scalatest which uses scalatestplus-junit yet.
In current newest release, JUnitRunner code is in scalatest directly:
https://github.com/scalatest/scalatest/blob/release-3.0.8/scalatest/src/main/scala/org/scalatest/junit/JUnitRunner.scala